turbo/stages/headerdownload/header_algos.go

```diff
@@ -878,38 +877,40 @@ func (hd *HeaderDownload) SetHeaderToDownloadPoS(hash common.Hash, height uint64
 	}
 }
 
-func (hd *HeaderDownload) ProcessSegmentPOS(segment ChainSegment, tx kv.Getter) error {
-	if len(segment) == 0 {
-		return nil
+func (hd *HeaderDownload) ProcessSegmentPOS(segment ChainSegment, tx kv.Getter, peerId [64]byte) ([]PenaltyItem, error) {
+	// segment of one header are usually remainder of gossip, but the peer can still be valid.
```

Comment:
Can you give an example when it happens? I'm a bit sceptical that we need this special check.